hcn/hcnnamespace.go

```diff
@@ -165,7 +165,10 @@ func createNamespace(settings string) (*HostComputeNamespace, error) {
 }
 
 func modifyNamespace(namespaceId string, settings string) (*HostComputeNamespace, error) {
-	namespaceGuid := guid.FromString(namespaceId)
+	namespaceGuid, err := guid.FromString(namespaceId)
```

Comment:
I think you should wrap all these in something that replaces a non-nil err with errors.New("invalid namespace ID") or something. Currently this is nearly undebuggable.

Reply:
(probably fmt.Errorf("invalid namespace ID %q", id))

Reply:
Would it be better to start using `errors.Wrap` more uniformly? Each of these `return nil, err` could instead be `return nil, errors.Wrap(err, "failed to parse namespace")`.

Reply:
I think in this case there is not much value of passing through the lower error.

In any case, I think having a helper function to do this would be better so that you can choose an appropriate error in one place.

Reply:
Replaced with a new error that just states a bad ID was passed in. I made separate errors for namespace ID, network ID, endpoint ID, and load balancer ID. I think this will help keep the error message actionable.